modules/AOW_Actions/FormulaCalculator/FormulaCalculatorPluginLoader.php

```diff
@@ -0,0 +1,187 @@
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for  technical reasons, the Appropriate Legal Notices must
+ * display the words  "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+/**
+ * Created by Adam Jakab.
+ * Date: 01/12/17
+ * Time: 15.40
+ */
+
+/**
+ * A temporary autoloader for FormulaCalculator until the day we will be using composer's autoloader
+ *
+ * Class FormulaCalculatorPluginLoader
+ */
+class FormulaCalculatorPluginLoader
+{
+    /** @var string */
+    protected static $projectRootPath;
+    
+    /** @var array */
+    protected static $plugins = Array();
+    
+    /**
+     *
+     */
+    public static function initialize()
+    {
+        self::$projectRootPath = dirname(dirname(dirname(__DIR__)));
+        
+        require_once(self::$projectRootPath
+            . '/modules/AOW_Actions/FormulaCalculator/Plugins/FormulaCalculatorBasePlugin.php');
+        require_once(self::$projectRootPath
+            . '/modules/AOW_Actions/FormulaCalculator/Plugins/FormulaCalculatorPluginInterface.php');
+        
+        spl_autoload_register("FormulaCalculatorPluginLoader::load");
+    }
+    
+    /**
+     * @param string $function
+     *
+     * @throws \Exception
+     *
+     * @return \AOW_Actions\FormulaCalculator\Plugins\FormulaCalculatorPluginInterface
+     */
+    public static function getPluginInstanceForFunction($function)
+    {
+        if (!array_key_exists($function, self::$plugins))
+        {
+            if (!$function)
+            {
+                throw new \Exception(__CLASS__ . ": No function name provided!");
+            }
+    
+            $fqcn = null;
+            $className = 'FormulaCalculator' . ucfirst($function) . 'Plugin';
+            
+            $fqcnTests = [];
+            $fqcnTests["custom"] = '\Custom\AOW_Actions\FormulaCalculator\Plugins\\' . $className;
+            $fqcnTests["core"] = '\AOW_Actions\FormulaCalculator\Plugins\\' . $className;
+            
+    
+            foreach ($fqcnTests as $key => $class)
+            {
+                if (class_exists($class))
+                {
+                    $fqcn = $class;
+                }
+            }
+            
+            if(!$fqcn)
+            {
+                throw new \Exception(__CLASS__ . ": Plugin class was not found for function(".$function.")! ");
+            }
+            
+            $reflection = new ReflectionClass($fqcn);
+            
+            if (!$reflection->implementsInterface("\AOW_Actions\FormulaCalculator\Plugins\FormulaCalculatorPluginInterface"))
+            {
+                throw new \Exception(__CLASS__
```

Comment:
So we are moving to an exception classes using PSR-4. See lib/Exception

Reply:
OK